src/clients/victoriaMetrics/promMetricGauge.ts

```diff
@@ -1,44 +1,56 @@
 import promClient, { Gauge } from 'prom-client';
 import { promAgent } from './promAgent';
-import { generateLabelCombinations } from './metricFunctions';
 
-interface GaugeConfig {
+export interface GaugeConfig {
   name: string;
   help: string;
-  labelNames?: string[];
-  labelOptions?: Record<string, string[]>;
+  labelOptions?: Record<string, Set<string>>;
   collect?: () => void; // Refer to prom-client docs on how this should be used.
 }
 
+export interface PromClientGaugeConfig {
+  name: string;
+  help: string;
+  labelNames?: string[];
+  collect?: () => void;
+}
+
 export class PromMetricGauge {
```

Comment:
Should we have our own set() method here so we can enforce labels are properly used too?

Reply:
Not easily possible as the collect function is to be declared before the metric is made

Reply:
Proved you wrong ;)